test/e2e/framework/framework.go

```diff
@@ -93,6 +93,11 @@ type TestDataSummary interface {
 type FrameworkOptions struct {
 	ClientQPS   float32
 	ClientBurst int
+
+	// Tie the framework to a specific node (e.g. for node e2e tests).
+	// NOTE: This must be a pointer because the node-name flag has not been parsed yet when the
+	// framework options are set.
```

Comment:
One of the goals is to be able to share tests between node and cluster level e2es. With this model, one will have to inject the framework into the tests. 
Instead, would it be possible to use `TestContext` to pass the NodeName as a flag and have these tests use a framework method for spawning pods? That way the tests can be re-used as is.